packages/destination-actions/src/destinations/intercom/identifyContact/index.ts

```diff
@@ -0,0 +1,187 @@
+  defaultSubscription: 'type = "identify"',
+  platform: 'web',
+  fields: {
+    role: {
+      type: 'string',
+      required: true,
+      description: 'The role of the contact. Accepted values are `user` or `lead`. Can only be updated if `lead`.',
+      label: 'Role',
+      default: 'lead'
+    },
+    external_id: {
+      type: 'string',
+      description: 'A unique identifier generated outside Intercom. Required if role=user and email is blank.',
+      label: 'External ID',
+      default: {
+        '@path': '$.userId'
+      }
+    },
+    email: {
+      type: 'string',
+      description: "The contact's email. Required if role=user and external_id is blank.",
+      label: 'Email',
+      format: 'email',
+      default: {
+        '@path': '$.traits.email'
+      }
+    },
+    phone: {
+      label: 'Phone Number',
+      description: "The contact's phone number.",
+      type: 'string',
+      default: {
+        '@path': '$.traits.phone'
+      }
+    },
+    name: {
+      type: 'string',
+      description: "The contact's name.",
+      label: 'Name',
+      default: {
+        '@path': '$.traits.name'
+      }
+    },
+    avatar: {
+      label: 'Avatar',
+      description: 'URL of image to be associated with contact profile.',
+      type: 'string',
+      format: 'uri',
+      default: {
+        '@path': '$.traits.avatar'
+      }
+    },
+    signed_up_at: {
+      label: 'Signed Up At',
+      type: 'datetime',
+      description: 'The timestamp when the contact was created.',
+      default: {
+        '@path': '$.createdAt'
+      }
+    },
+    last_seen_at: {
+      label: 'Timestamp',
+      type: 'datetime',
+      description: 'The timestamp the contact was last seen.',
+      default: {
+        '@path': '$.timestamp'
+      }
+    },
+    owner_id: {
+      label: 'Owner Id',
+      type: 'number',
+      description: 'The id of an admin that has been assigned account ownership of the contact.'
+    },
+    unsubscribed_from_emails: {
+      label: 'Unsubscribed From Emails',
+      type: 'boolean',
+      description: 'Whether the contact is unsubscribed from emails.'
+    },
+    custom_attributes: {
+      label: 'Custom Attributes',
+      description:
+        'The custom attributes which are set for the contact. Note: Will throw an error if the object has an attribute that isn`t explicitly defined on Intercom.',
+      type: 'object',
+      default: {
+        '@path': '$.traits.customAttributes'
+      }
+    }
+  },
+  perform: async (request, { payload }) => {
+    /**
+     * Tries to search and update the contact first. If no contact is found, then create.
+     * This is because we anticipate many more updates than creations happening in practice.
+     *
+     * Note: When creating a lead, Intercom doesn't accept an external_id (?), it only accepts email
+     */
+    try {
+      const contact = await searchIntercomContact(request, payload)
+      if (contact) {
```

Comment:
I think you can remove the `else` clause since `contact` will always be defined.
```suggestion
      if (contact) {
              return updateIntercomContact(request, contact.id, payload)
      }
      return await createIntercomContact(request, payload)
```

Reply:
Discussed on slack - `searchIntercomContact` does return `undefined`, but we should get rid of the else regardless because explicit else cases are not helpful unless they provide value :)